cartography/data/jobs/cleanup/aws_import_lambda_cleanup.json

```diff
@@ -0,0 +1,15 @@
+{
+    "statements": [
+        {
+            "query": "MATCH (:AWSAccount{id: {AWS_ID}})-[:RESOURCE]->(n:AWSLambda) WHERE n.lastupdated <> {UPDATE_TAG} WITH n LIMIT {LIMIT_SIZE} DETACH DELETE (n) return COUNT(*) as TotalCompleted",
+            "iterative": true,
+            "iterationsize": 100
+        },
+        {
+            "query": "MATCH (a:AWSLambda) WHERE NOT (a)-[:STS_ASSUME_ROLE_ALLOW]->(:AWSPrincipal) WITH a LIMIT {LIMIT_SIZE} DETACH DELETE (a) return COUNT(*) as TotalCompleted",
```

Comment:
Change to `MATCH (:AWSLambda)-[r:STS_ASSUME_ROLE_ALLOW]->(:AWSPrincipal) WITH a LIMIT {LIMIT_SIZE} DETACH DELETE (a) return COUNT(*) as TotalCompleted"`.

I think the cleanup job is good like this unless I missed something.